iofog-agent-daemon/src/test/java/org/eclipse/iofog/field_agent/FieldAgentTest.java

```diff
@@ -0,0 +1,1605 @@
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void initiateMockStart() {
+        thread = mock(Thread.class);
+        try {
+            whenNew(Thread.class).withParameterTypes(Runnable.class,String.class).withArguments(any(Runnable.class),
+                    anyString()).thenReturn(thread);
+            PowerMockito.doNothing().when(thread).start();
+            fieldAgent.start();
+        } catch (Exception e) {
+            fail("this should not happen");
+        }
+
+    }
+
+    /**
+     * Test module index of FieldAgent
+     */
+    @Test
+    public void testGetModuleIndex() {
+        assertEquals(5, fieldAgent.getModuleIndex());
+    }
+
+    /**
+     * Test Module Name of FieldAgent
+     */
+    @Test
+    public void getModuleName() {
+        assertEquals(MODULE_NAME, fieldAgent.getModuleName());
+
+    }
+
+    /**
+     * Test getInstance is same as mock
+     */
+    @Test
+    public void testGetInstanceIsSameAsMock() {
+        assertSame(fieldAgent, FieldAgent.getInstance());
+    }
+
+    /**
+     * Test postTracking with valid jsonObject
+     */
+    @Test ( timeout = 5000L )
+    public void testPostTrackingWithValidJsonObject() {
+        try {
+            initiateMockStart();
+            fieldAgent.postTracking(jsonObject);
+            Mockito.verify(orchestrator).request(any(), any(), any(), any());
+            PowerMockito.verifyStatic(LoggingService.class, Mockito.never());
+            LoggingService.logError(eq(MODULE_NAME), eq("Unable send tracking logs"), any());
+        } catch (Exception e) {
+            fail("This should not happen");
+        }
+    }
+
+    /**
+     * Test postTracking with null jsonObject
+     */
+    @Test ( timeout = 5000L )
+    public void throwsExceptionPostTrackingWithNullJsonObject() {
+        try {
+            initiateMockStart();
+            PowerMockito.when(orchestrator.request(any(), any(), any(), any())).thenThrow(mock(Exception.class));
+            fieldAgent.postTracking(null);
+            Mockito.verify(orchestrator).request(any(), any(), any(), any());
+            PowerMockito.verifyStatic(LoggingService.class, Mockito.atLeastOnce());
+            LoggingService.logError(eq(MODULE_NAME), eq("Unable send tracking logs"), any());
+        } catch (Exception e) {
+            fail("this should never happen");
+        }
+    }
+
+    /**
+     * Test provision when controller status is not provisioned
+     */
+    @Test
+    public void testProvisionWhenControllerStatusIsNotProvisioned() {
+        try {
+            initiateMockStart();
+            PowerMockito.when(orchestrator.provision(anyString())).thenReturn(provisionJsonObject);
+            JsonObject response = fieldAgent.provision("provisonKey");
+            assertTrue(response.containsKey("message"));
+            assertEquals("success", response.getString("message"));
+            Mockito.verify(orchestrator).provision(anyString());
```

Comment:
verify that `orchestrator.provision` gets called with "provisionKey"

Reply:
updated the verification